tests/byoc/byoc.go

```diff
@@ -0,0 +1,260 @@
+
+	// Initialize the application struct
+	application := &appservice.Application{}
+	cdq := &appservice.ComponentDetectionQuery{}
+	compDetected := appservice.ComponentDetectionDescription{}
+	componentObj := &appservice.Component{}
+
+	for _, suite := range byocTestScenario {
+		suite := suite
+
+		Describe(suite.Name, func() {
+			BeforeAll(func() {
+				fw, err = framework.NewFramework(utils.GetGeneratedNamespace(fmt.Sprintf("byoc-%s", strings.ToLower(string(suite.Byoc.ClusterType)))))
+				Expect(err).NotTo(HaveOccurred())
+
+				if suite.Byoc.ClusterType == appservice.ConfigurationClusterType_Kubernetes {
+					vc = vcluster.NewVclusterController(fmt.Sprintf("%s/tmp", rootPath), fw.AsKubeAdmin.CommonController.CustomClient)
+				}
+			})
+
+			// Remove all resources created by the tests
+			AfterAll(func() {
+				if !CurrentSpecReport().Failed() {
+					Expect(fw.AsKubeDeveloper.HasController.DeleteAllComponentsInASpecificNamespace(fw.UserNamespace, 30*time.Second)).To(Succeed())
+					Expect(fw.AsKubeAdmin.HasController.DeleteAllApplicationsInASpecificNamespace(fw.UserNamespace, 30*time.Second)).To(Succeed())
+					Expect(fw.AsKubeAdmin.HasController.DeleteAllSnapshotEnvBindingsInASpecificNamespace(fw.UserNamespace, 30*time.Second)).To(Succeed())
+					Expect(fw.AsKubeAdmin.ReleaseController.DeleteAllSnapshotsInASpecificNamespace(fw.UserNamespace, 30*time.Second)).To(Succeed())
+					Expect(fw.AsKubeAdmin.GitOpsController.DeleteAllEnvironmentsInASpecificNamespace(fw.UserNamespace, 30*time.Second)).To(Succeed())
+					Expect(fw.AsKubeAdmin.TektonController.DeleteAllPipelineRunsInASpecificNamespace(fw.UserNamespace)).To(Succeed())
+					Expect(fw.AsKubeAdmin.GitOpsController.DeleteAllGitOpsDeploymentInASpecificNamespace(fw.UserNamespace, 30*time.Second)).To(Succeed())
+					Expect(fw.SandboxController.DeleteUserSignup(fw.UserName)).NotTo(BeFalse())
+				}
+			})
+
+			It("initializes byoc cluster connection", func() {
+				if suite.Byoc.ClusterType == appservice.ConfigurationClusterType_Kubernetes {
+					byocKubeconfig, err = vc.InitializeVCluster(fw.UserNamespace, fw.UserNamespace)
+					Expect(byocKubeconfig).NotTo(BeEmpty(), "failed to initialize vcluster. Kubeconfig not provided")
+					Expect(err).NotTo(HaveOccurred())
+
+				} else if suite.Byoc.ClusterType == appservice.ConfigurationClusterType_OpenShift {
+					byocKubeconfig = utils.GetEnv("BYOC_KUBECONFIG", "")
+					Expect(byocKubeconfig).NotTo(BeEmpty(), "Please provide BYOC_KUBECONFIG env pointing to a valid openshift kubeconfig file")
+				}
+
+				config, err := clientcmd.BuildConfigFromFlags("", byocKubeconfig)
+
+				Expect(err).NotTo(HaveOccurred())
+				byocAPIServerURL = config.Host
+				Expect(byocAPIServerURL).NotTo(BeEmpty())
+
+				ephemeralClusterClient, err = client.NewKubeFromKubeConfigFile(byocKubeconfig)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("creates managed-gitops.redhat.com/managed-environment secret type", func() {
```

Comment:
Is this more of a precondition?

Reply:
Is a step to import your cluster to RHTAP